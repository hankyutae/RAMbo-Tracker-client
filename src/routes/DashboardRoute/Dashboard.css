.user-dashboard{
  display: flex;
  flex-direction: column;
}

#dash-button-container {
  display: inherit;
  flex-direction: row;
  align-content: space-between;
  width: 50%;
  margin: 0 auto;
  padding-bottom: 2%;
}
#user-welcome {
  text-align: left;
  padding-left: 3%;
}
/* #user-name{
  color: blue;
  
} */
/* #meal-button{
  background-color: rgb(255, 123, 0);;
}
#symptom-button{
  background-color: rgb(54, 1, 54);
} */
.meal{
  background-color: #49797B;
}
.symptom{
  background-color: #7B4A48;
}
.events > li {
  color:white;
  border: 2px solid black;
  width:fit-content;
  
}
li {
  list-style-type: none;
  color: white;
  padding: 2%;
  
}
.events-list {
  width: 50%;
  margin: 0 auto;
}

.Modal{
  display: flex;
  position: absolute;
  top: 40px;
  left: 40px;
  right: 40px;
  bottom: 40px;
  background-color: white;
  
}

.Overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgb(224, 224, 224); 
  
} 

.symptom-modal{
  opacity: 0.5;
}

.dashboard-content {
  background-image: url('../../Media/dash-background.jpg');
  background-size: cover;
  padding: 5%;
  margin: 0 5%;
}
