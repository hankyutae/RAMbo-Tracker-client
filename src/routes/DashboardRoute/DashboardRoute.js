import React from "react";
import API from "../../services/api-service";
import Modal from "react-modal";
//components
import Symptoms from "../../components/Symptom/Symptom";
import Meal from "../MealRoute/MealRoute";
//css
import "./Dashboard.css";
import Result from "../../components/Result/Result";
import Plate from "../../Media/plate.png";
import Symptom from "../../Media/symptom.png";
import Printer from '../../Media/print.png'
//to be removed for final product
//import helper from "../../services/helper.services";

if (process.env.NODE_ENV !== "test") Modal.setAppElement("#root");
export default class DashBoard extends React.Component {
  state = {
    addMealModal: false,
    addSymptomsModal: false,
    expanded: false,
    itemExpanded: [],
    forceUpdateInResult:false,

    user: {
      username: "",
      display_name: ""
    },
    events: [
      {
        name: "",
        time: 2134234,
        type: "",
        items: [
          {
            name: "",
            ingredients: []
          },
          {
            name: "",
            ingredients: []
          }
        ]
      }
    ]
  };
  clearErrors=()=>{
    this.setState({error: null})
  }
  componentDidMount() {
    this.clearErrors()
    API.doFetch("/event")
      .then(res => {
        this.setState({
          user: { username: res.username, display_name: res.display_name },
          events: res.events
        });
      })
      .catch(res => this.setState({ error: res.message }));
  }

  updateAllEventsDueToResult= ()=>{
    this.clearErrors()
    API.doFetch("/event")
      .then(res => {
        this.setState({
          user: { username: res.username, display_name: res.display_name },
          events: res.events
        });
      })
      .catch(res => this.setState({ error: res.message }));
  }

  handleDelete = (id, type, index) => {
    this.clearErrors()

    API.doFetch("/event", "DELETE", {
      id,
      type
    }).then(() => {
      const newEvents = [...this.state.events];
      newEvents.splice(index, 1);
      this.setState({
        events: newEvents,
        forceUpdateInResult:!this.state.forceUpdateInResult
      });
    })
    .catch(res =>this.setState({error: res.message}));
  };

  handleExpandToggle = index => {
    if (this.state.expanded === index) {
      this.setState({
        expanded: false,
        itemExpanded: []
      });
    } else {
      this.setState({
        expanded: index,
        itemExpanded: []
      });
    }
  };

  handleIngredientsToggle = index => {
    if (this.state.itemExpanded.includes(index)) {
      const newItemExpanded = [...this.state.itemExpanded];
      newItemExpanded.splice(newItemExpanded.indexOf(index), 1);
      this.setState({
        itemExpanded: newItemExpanded
      });
    } else {
      const newItemExpanded = [...this.state.itemExpanded];
      newItemExpanded.push(index);
      this.setState({
        itemExpanded: newItemExpanded
      });
    }
  };

  closeModal = modal => {
    this.setState({ [modal]: false });
  };
  openModal = (e, modal) => {
    e.preventDefault();
    this.setState({ [modal]: true });
  };
  updateEvents = e => {
    let temp = [e, ...this.state.events];
    temp.sort(
      (a, b) => new Date(b.time).getTime() - new Date(a.time).getTime()
    );
    this.setState({ events: temp,
      forceUpdateInResult:!this.state.forceUpdateInResult });
  };
  formatDate = time => {
    let date = new Date(time);
    let formatted_date =
      date.getMonth() +
      1 +
      "-" +
      date.getDate() +
      "-" +
      date.getFullYear() +
      " " +
      date.getHours() +
      ":" +
      date.getMinutes();
    return formatted_date;
  };
 
  print = () => { 
     window.print() 
  };
  render() {
    let printEvents =
    this.state.events.map((e, index) => {
      if (e.type === "meal") {
        return (
          <div key={index} className="print-container">
            <li className="meal">
              {e.name} at {this.formatDate(e.time)}
              <ul className="food-ingredient-print">
                {e.items.map((item, index) => {
                  return (
                    <li key={index} className="food-item-in-dash">
                      <p className="food-info-in-dash">{item.name}:{' '}
                        {item.ingredients
                          .map(ingredient => ingredient.toLowerCase())
                          .join(", ")}
                          
                      </p>
                    </li>
                  );
                })}
              </ul>
              
            </li>
          </div>
        );
      } else{
        return (
          <div key={index} className="print-container">
            <li className="symptom">
              {e.name} at {this.formatDate(e.time)}{" "}
              {e.type === "symptom" ? `Severity: ${e.severity}` : ""}
            </li>
          </div>
        );
      }
    });
    let events = this.state.events.map((e, index) => {
      if (e.type === "meal") {
        return (
          <div key={index} className="dash-event-container-meal">
            <li className={"meal"}>
              {e.name} at {this.formatDate(e.time)}
              <div className='meal-toggle-cont'>
              <button
                className="expand-toggle"
                onClick={() => this.handleExpandToggle(index)}
              >
                {this.state.expanded === index ? "-" : "+"}
              </button>
              <button
                className="delete-event"
                onClick={() => this.handleDelete(e.id, e.type, index)}
              >
                <i className="fa fa-trash" aria-hidden="true"></i>
              </button>
              </div>
              {this.state.expanded === index && (
                <ul className="food-toggle">
                  {e.items.map((item, index) => {
                    return (<>
                      <li key={index} className="food-item-in-dash">
                        <p className="food-info-in-dash">{item.name}</p>
                        <p className="ingredients-list-in-dash">
                          {this.state.itemExpanded.includes(index) &&
                            item.ingredients
                              .map(ingredient => ingredient.toLowerCase())
                              .join(", ")}
                          
                        </p>
                      </li>
                      <div className='exp-hide-btn'>
                          <button
                            className="ingredients-expand"
                            onClick={() => this.handleIngredientsToggle(index)}
                          >
                            {this.state.itemExpanded.includes(index)
                              ? "Hide ingredients"
                              : "Show ingredients"}
                          </button>
                          </div>
                    </>);
                  })}
                </ul>
              )}
            </li>
          </div>
        );
      } else {
        return (
          <div key={index} className="dash-event-container-symptom">
            <li className="symptom">
              {e.name} at {this.formatDate(e.time)}{" "}
              {e.type === "symptom" ? `Severity: ${e.severity}` : ""}{' '}
              <button
                className="delete-event"
                onClick={() => this.handleDelete(e.id, e.type, index)}
              >
                <i className="fa fa-trash" aria-hidden="true"></i>
              </button>
            </li>
           
          </div>
        );
      }
    });
    return (
      <div>
        {/*add meal modal*/}
        <Modal
          isOpen={this.state.addMealModal}
          onRequestClose={e => this.closeModal("addMealModal")}
        >
          <Meal closeModal={this.closeModal} updateEvents={this.updateEvents} />
        </Modal>
        <Modal
          isOpen={this.state.addSymptomsModal}
          onRequestClose={() => this.closeModal("addSymptomsModal")}
        >
          <Symptoms
            closeModal={this.closeModal}
            prevSymptoms={this.state.events.filter(e => e.type === "symptom")}
            updateEvents={this.updateEvents}
          />
        </Modal>
        <div id="user-welcome">
          {" "}
          <h3>Welcome back, {this.state.user.display_name}</h3>
        </div>
        <div className="dashboard-content">
          <Result refreshDash={this.updateAllEventsDueToResult} forceUpdate={this.state.forceUpdateInResult}/>
          <div className="log-container">
            <h2>My Log</h2>
            {this.state.error && <p className="error">There Was An Error!</p>}

            <div id="dash-button-container">
              <button
                className="user-button new-meal"
                onClick={e => this.openModal(e, "addMealModal")}
              >
                <img className="button-logo" src={Plate} alt=""></img>
                Log Meal
              </button>
              <button
                className="user-button new-symptom"
                onClick={e => this.openModal(e, "addSymptomsModal")}
              >
                <img className="button-logo" src={Symptom} alt=""></img>
                Log Symptom
              </button>
          <button className="user-button print-button" onClick={() => this.print()}>
            <img className="button-logo" src={Printer} alt=""></img>
            Print Logs
          </button>
          <div className=".print-container">

        {printEvents}
          </div>
            </div>
            <div className="events">
              <div className="events-list">
                {events == "" ? `Your log is empty` : events}
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
