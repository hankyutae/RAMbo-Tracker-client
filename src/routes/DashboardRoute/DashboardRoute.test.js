import React from 'react';
import ReactDOM from 'react-dom';
import DashboardRoute from './DashboardRoute';
import { BrowserRouter } from 'react-router-dom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <DashboardRoute />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});