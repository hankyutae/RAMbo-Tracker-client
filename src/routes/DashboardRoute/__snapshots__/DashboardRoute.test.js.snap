// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <div
    id="user-welcome"
  >
     
    <h3>
      Welcome back, 
      
    </h3>
    <div
      className="results"
    >
      <ul>
        <p>
          click refresh to see correlation
        </p>
      </ul>
      <button
        onClick={[Function]}
      >
        refresh
      </button>
    </div>
  </div>
  <div
    className="dashboard-content"
  >
    <div
      id="dash-button-container"
    >
      <button
        className="user-button"
        onClick={[Function]}
      >
        Log New Meal
      </button>
      <button
        className="user-button"
        onClick={[Function]}
      >
        Log New Symptoms
      </button>
    </div>
    <div
      className="events"
    >
      <div
        className="events-list"
      >
        <li
          className="symptom"
        >
          
           at 
          Wed Dec 31 1969
           
        </li>
      </div>
    </div>
  </div>
</div>
`;
